numpy/core/src/multiarray/alloc.c

```diff
@@ -113,8 +113,11 @@ npy_free_cache(void * p, npy_uintp sz)
 NPY_NO_EXPORT void *
 npy_alloc_cache_dim(npy_uintp sz)
 {
-    /* dims + strides */
-    if (NPY_UNLIKELY(sz < 2)) {
+    /*
+     * make sure any temporary allocation can be used for array metadata which
+     * uses one memory block for both dimensions and strides
+     */
```

Comment:
Right, but before this PR that was the only cache path.

So is this really a performance optimization so that 0d arrays preallocate data for 1d arrays? Or is this just working around calling `malloc(0)`, which might return `NULL`?